crates/core/tedge_agent/src/agent.rs

```diff
@@ -184,7 +189,20 @@ impl Agent {
         let signal_actor_builder = SignalActor::builder(&runtime.get_handle());
 
         // Health actor
-        let health_actor = HealthMonitorBuilder::new(TEDGE_AGENT, &mut mqtt_actor_builder);
+        // TODO: take a user-configurable service topic id
+        let service_topic_id = self.config.mqtt_device_topic_id.to_default_service_topic_id("tedge-agent")
+            .with_context(|| format!("Device topic id {} currently needs default scheme, e.g: 'device/DEVICE_NAME//'", self.config.mqtt_device_topic_id))?;
+        let service = Service {
+            service_topic_id,
+            device_topic_id: DeviceTopicId::new(self.config.mqtt_device_topic_id.clone()),
+        };
+        let mqtt_schema = MqttSchema::with_root(self.config.mqtt_topic_root.to_string());
+        let health_actor = HealthMonitorBuilder::from_service_topic_id(
+            service,
+            &mut mqtt_actor_builder,
+            &mqtt_schema,
+            self.config.service_type.clone(),
+        );
```

Comment:
This is done again and again for all tedge daemons. I would consider to add a specific `HealthMonitorBuilder::from_config()` constructor.